File: python/grammer/03_string_slicing.py
"""3) 문자열 슬라이싱: 시작/끝/간격 사용법."""

word = "algorithm"

print("원본:", word)
print("word[1:5] ->", word[1:5])      # 1번부터 4번 인덱스까지
print("word[:4]  ->", word[:4])       # 처음부터 3번 인덱스까지
print("word[::2] ->", word[::2])      # 2칸씩 건너뛰기
print("word[::-1] ->", word[::-1])    # 뒤집기

File: cpp/grammer/03_string_slicing.cpp
#include <iostream>
#include <string>

int main() {
    std::string word = "algorithm";
    std::cout << "원본: " << word << "\n";
    std::cout << "word[1:5] -> " << word.substr(1, 4) << "\n";
    std::cout << "word[:4] -> " << word.substr(0, 4) << "\n";

    std::string step;
    for (size_t i = 0; i < word.size(); i += 2) step += word[i];
    std::cout << "word[::2] -> " << step << "\n";

    std::string rev(word.rbegin(), word.rend());
    std::cout << "word[::-1] -> " << rev << "\n";
}
